src/OrchardCore.Modules/OrchardCore.Templates/Services/AdminTemplatesShapeBindingResolver.cs

```diff
@@ -65,6 +68,36 @@ public async Task<ShapeBinding> GetShapeBindingAsync(string shapeType)
             }
         }
 
+        public async Task<IEnumerable<string>> GetShapeBindingNamesAsync(Func<string, bool> predicate, bool adminTemplate)
```

Comment:
I don't think we should put a property here for `adminTemplate`

It makes things too tightly couple to the concept of admintemplates vs ordinary templates.

Possibly better is to put a marker interface on the class, `IAdminTemplateShapeBindingResolver` and register it as such.

Then when you need admin templates, resolve the enumerables for `IAdminTemplateShapeBindingResolver`

But when I look at how shape binding resolvers are used, I don't quite understand why you are trying to differentiate between them at all? We aren't doing any differentiation on display, so maybe you can explain why we need to for names?

Reply:
We can get the shape table by theme id and shape names from that shape table. However currently it's not possible to get shape names for dynamic shapes that are added by `IShapeBindingResolver` 

For dynamic templates we have only two kind of dynamic templates UI and Admin Templates. 
The Template module implementation has two separate `IShapeBindingResolver` for each kind.

However for `IShapeBindingResolver`  developer could define custom logic to provide shapes (for admin and UI both in single implementation).  For example, I could write `IShapeBindingResolver` that reads template from a json file or single blob storage or git repo that provide both UI and Admin shapes (see my failed attempt to replace template manager #7001,  and i decided implement my own `IShapeBindingResolver`)  

Idea is to get the list of shape names from the `IShapeBindingResolver` and based on provided filter condition. 
For now in this PR , I use this new method to get shape names in `EditTypePart.cshtml`.  

However it can be useful in future to get list of UI shape names for some future module.

Reply:
Yep, cool, I'm getting what you added the method for.

I don't think it should have this property on it :`, bool adminTemplate)`

Reply:
May be we can have two methods one for each?

Reply:
Sorry, didn't see this last comment from you last week. 

Thinking 🤔

Reply:
```
public AdminTemplatesShapeBindingResolver : IShapeBindingResolver, IAdminTemplatesShapeBindingNameResolver
public TemplatesShapeBindingResolver : IShapeBindingResolver, ITemplatesShapeBindingNameResolver

services.AddScoped<IAdminTemplatesShapeBindingNameResolver, AdminTemplatesShapeBindingResolver>()
services.AddScoped<IShapeBindingResolver>(sp => sp.GetRequiredService<IAdminTemplatesShapeBindingResolver>();
```

Then in your view for edit `@inject IEnumerable<IAdminTemplatesShapeBindingNameResolver> ShapeBindingResolvers`
or for the other `@inject IEnumerable<ITemplatesShapeBindingNameResolver>`

or similar. does that make sense?

and keep the `public async Task<IEnumerable<string>> GetShapeBindingNamesAsync` as a method on the `IAdminTemplatesShapeBindingNameResolver` interface, not on the `IShapeBindingResolver` (probably)

Bit of extra code, and more indirection, but keeps the implementations a bit cleaner

Reply:
Sounds good! I'll make the change